src/UuidAggregateIdTrait.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lendable\Aggregate;
+
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * Universally unique identifier for an aggregate.
+ */
+trait UuidAggregateIdTrait
+{
+    abstract public static function generate(): self;
+
+    private function __construct(protected readonly UuidInterface $uuid)
+    {
+    }
+
+    public static function fromString(string $uuid): self
+    {
+        return new self(Uuid::fromString($uuid));
+    }
+
+    public static function fromBinary(string $binaryUuid): self
+    {
+        return new self(Uuid::fromBytes($binaryUuid));
+    }
+
+    public static function fromUuid(UuidInterface $uuid): self
+    {
+        return new self($uuid);
+    }
+
+    public function toString(): string
+    {
+        return $this->uuid->toString();
+    }
+
+    public function toBinary(): string
+    {
+        return $this->uuid->getBytes();
+    }
+
+    public function equals(AggregateId $other): bool
+    {
+        return $this->uuid->toString() === $other->toString();
```

Comment:
```suggestion
        return $this->uuid->equals($other->uuid);
```

Reply:
Alternatively

```suggestion
        return $other instanceof $this && $this->uuid->equals($other->uuid);
```

But this comes down to semantics of if `FooId` and `BarId` with the same underlying UUID are equal? I would say they are not.

Reply:
I'm not sure if I can easily make use of equals.
Check is done from trait, so I can't have access to field, we have to expose uuid instance on the interface to make it work?.

Reply:
Like this:
https://github.com/Lendable/aggregate/pull/566/commits/91d6c6be1788edadbab47dc6479c2e1283fee014

Reply:
As the trait is mixed in to the actual class it has full private member access of other instances of the same type, so it works for me (tests pass) with this change locally:

```php
    public function equals(AggregateId $other): bool
    {
        return $other instanceof $this
            && $this->uuid->equals($other->uuid);
    }
```

Reply:
I was sure that phpstan had a problem that make sense yesterday, but maybe I something else was wrong 🤔 
Passes now, happy to change